trio/_core/_io_windows.py

```diff
@@ -724,12 +748,14 @@ def notify_closing(self, handle):
     ################################################################
 
     @_public
-    def register_with_iocp(self, handle):
+    def register_with_iocp(self, handle: int | CData) -> None:
         self._register_with_iocp(handle, CKeys.WAIT_OVERLAPPED)
 
     @_public
-    async def wait_overlapped(self, handle, lpOverlapped):
-        handle = _handle(handle)
+    async def wait_overlapped(
+        self, handle_: int | CData, lpOverlapped: CData | int
```

Comment:
Instead of casting to/from `CData`, maybe we should type it as our `Overlapped` / `Handle` types, so this can be statically checked? Or just add those to the union alongside `CData`, to be more loose.

Reply:
I would push out `lpOverlapped` casting to a boundary... but looking at the code, this is public :(

The type casting can probably be improved though.